File: performance-comparison/benchmarks/io/python/file_read.py
import sys, os

n    = int(sys.argv[1]) if len(sys.argv) > 1 else 1048576
path = "/tmp/bench_io_read_py.bin"
with open(path, 'wb') as fw:
    rem = n
    while rem > 0:
        chunk = min(4096, rem)
        fw.write(b'\xCD' * chunk)
        rem -= chunk
total = 0
with open(path, 'rb') as fr:
    while True:
        data = fr.read(4096)
        if not data:
            break
        total += len(data)
os.remove(path)
print(total)

File: performance-comparison/benchmarks/io/clojure/file_read.clj
(let [n    (Long/parseLong (first *command-line-args*))
     path  "/tmp/bench_io_read_clj.bin"
     wbuf  (byte-array 4096 (unchecked-byte 0xCD))]
  ;; write phase
  (with-open [fw (java.io.FileOutputStream. path)]
    (loop [rem n]
      (when (> rem 0)
        (let [chunk (min 4096 rem)]
          (.write fw wbuf 0 (int chunk))
          (recur (- rem chunk))))))
  ;; read phase
  (let [rbuf (byte-array 4096)
        total (with-open [fr (java.io.FileInputStream. path)]
                (loop [t 0]
                  (let [nr (.read fr rbuf)]
                    (if (< nr 0) t (recur (+ t nr))))))]
    (.delete (java.io.File. path))
    (println total)))
